plugins/octant/cmd/antrea-traceflow-plugin/main.go

```diff
@@ -0,0 +1,401 @@
+			dport, err = strconv.Atoi(dstPort)
+			if err != nil {
+				log.Printf("failed to get destination port: %s", err)
+				return nil
+			}
+		}
+		switch tf.Spec.Packet.IPHeader.Protocol {
+		case TCPProtocol:
+			{
+				tf.Spec.Packet.TransportHeader.TCP = &opsv1alpha1.TCPHeader{
+					SrcPort: int32(sport),
+					DstPort: int32(dport),
+				}
+			}
+		case UDPProtocol:
+			{
+				tf.Spec.Packet.TransportHeader.UDP = &opsv1alpha1.UDPHeader{
+					SrcPort: int32(sport),
+					DstPort: int32(dport),
+				}
+			}
+		case ICMPProtocol:
+			{
+				tf.Spec.Packet.TransportHeader.ICMP = &opsv1alpha1.ICMPEchoRequestHeader{
+					ID:       0,
+					Sequence: 0,
+				}
+			}
+		}
+		tf, err = a.client.OpsV1alpha1().Traceflows().Create(ctx, tf, metav1.CreateOptions{})
+		if err != nil {
+			log.Printf("failed to create traceflow CRD \"%s\", err: %s", tfName, err)
+			return nil
+		}
+		log.Printf("create traceflow CRD \"%s\" successfully, Traceflow Results: %+v", tfName, tf)
+		a.latestTfName = tf.Name
+		a.graph = graphviz.GenGraph(tf)
+		return nil
+	case showGraphAction:
+		name, err := request.Payload.String("name")
+		if err != nil {
+			return fmt.Errorf("unable to get name at string : %w", err)
+		}
+		// Invoke GenGraph to show
+		ctx := context.Background()
+		tf, err := a.client.OpsV1alpha1().Traceflows().Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			log.Printf("get traceflow CRD \"%s\" unsuccessfully, err: %s ", name, err)
+			return nil
+		}
+		log.Printf("get traceflow CRD \"%s\" successfully, Traceflow Results: %+v", name, tf)
+		a.latestTfName = tf.Name
+		a.graph = graphviz.GenGraph(tf)
+		return nil
+	default:
+		return fmt.Errorf("received action request for %s, but no handler defined", pluginName)
+	}
+}
+
+func (a *traceflowPlugin) initRoutes(router *service.Router) {
+	router.HandleFunc("/components", a.traceflowHandler)
+}
+
+func (a *traceflowPlugin) traceflowHandler(request service.Request) (component.ContentResponse, error) {
+	layout := flexlayout.New()
+	card := component.NewCard(component.TitleFromString("Antrea Traceflow"))
+	form := component.Form{Fields: []component.FormField{
+		component.NewFormFieldText(srcNamespaceCol, srcNamespaceCol, ""),
+		component.NewFormFieldText(srcPodCol, srcPodCol, ""),
+		component.NewFormFieldText(srcPortCol, srcPortCol, ""),
+		component.NewFormFieldText(dstNamespaceCol, dstNamespaceCol, ""),
+		component.NewFormFieldText(dstPodCol, dstPodCol, ""),
+		component.NewFormFieldText(dstPortCol, dstPortCol, ""),
+		component.NewFormFieldText(protocolCol, protocolCol, ""),
+		component.NewFormFieldHidden("action", addTfAction),
+	}}
+	addTf := component.Action{
+		Name:  "Start New Trace",
+		Title: "Start New Trace",
+		Form:  form,
+	}
+	graphForm := component.Form{Fields: []component.FormField{
+		component.NewFormFieldText("name", "name", ""),
+		component.NewFormFieldHidden("action", showGraphAction),
+	}}
+	genGraph := component.Action{
+		Name:  "Generate Trace Graph",
```

Comment:
Since we can support automatically refreshment for the latest traceflow, do you think it is better that we change this to " Generate History Trace Graph"?

Reply:
historical?

Reply:
> Since we can support automatically refreshment for the latest traceflow, do you think it is better that we change this to " Generate Historical Trace Graph"?

“LatestTfName” will change to not only the name of the most recently created traceflow, but also the name of the most recently shown traceflow. If so, we have to modify this part of logic.

Reply:
Do you mean "past"/"previous" instead of "historical"?

Reply:
I don't think we need to change this part of logic. It is just that "Start New Trace" will actually create a CRD and generate its trace graph. "Generate Historical Trace Graph" will generate the trace graph for any traceflow CRD given its name whether it is the latest one or not. BTW, do yo think maybe "lastTfName" can represent better than "latestTfName"? And we will always show graphviz for "lastTfName".

Reply:
> I don't think we need to change this part of logic. It is just that "Start New Trace" will actually create a CRD and generate its trace graph. "Generate Historical Trace Graph" will generate the trace graph for any traceflow CRD given its name whether it is the latest one or not. BTW, do yo think maybe "lastTfName" can represent better than "latestTfName"? And we will always show graphviz for "lastTfName".

"Latest" implies the most recently created one. As far as I'm concerned, we do need to change the name.